pkg/util/json/json.go

```diff
@@ -1013,6 +1016,48 @@ func (j jsonObject) Exists(s string) (bool, error) {
 	return v != nil, nil
 }
 
+func (j jsonNull) StripNulls() (JSON, error) {
+	return j, nil
+}
+func (j jsonTrue) StripNulls() (JSON, error) {
+	return j, nil
+}
+func (j jsonFalse) StripNulls() (JSON, error) {
+	return j, nil
+}
+func (j jsonNumber) StripNulls() (JSON, error) {
+	return j, nil
+}
+func (j jsonString) StripNulls() (JSON, error) {
+	return j, nil
+}
+func (j jsonArray) StripNulls() (JSON, error) {
+	var err error
+	jsons := make(jsonArray, len(j))
+	for i, e := range j {
+		if jsons[i], err = e.StripNulls(); err != nil {
+			return nil, err
+		}
+	}
+	return jsons, nil
+}
+func (j jsonObject) StripNulls() (JSON, error) {
```

Comment:
You might have considered this, but I think it's worth thinking about a case like this:
```
jsonb_strip_nulls({a: { ...very big object without any null values... }, b: null})
```
Ideally we would like to not allocate a new object for the value of `a`. We could accomplish this by having `StripNulls` (or an internal helper) pass up whether it did any replacements or not, and if not, just pass the original object up without re-allocating.

The implementation of this might look like iterating through the object until we either hit an actual replacement (in which case we need to copy the entire previous object over to a new one and then proceed as in the existing implementation) or hit the end (in which case we can just return the input).

Does that seem reasonable? An implementation like this would certainly be more complex and would probably require more substantial testing, but it might give some significant wins in many common cases (although if the implementation turns out too complex it might not be worth the trouble).